Amazon.QLDB.Driver/session/QldbSession.cs

```diff
@@ -170,6 +170,11 @@ public T Execute<T>(Func<TransactionExecutor, T> func)
 
                 throw ase;
             }
+            catch (Exception e)
+            {
+                this.NoThrowAbort(transaction);
+                throw e;
```

Comment:
Should we update the docstrings of this method and include the non-retriable exception(s) under the exception section?